cordformation/src/main/kotlin/net/corda/plugins/Dockerform.kt

```diff
@@ -84,13 +99,101 @@ open class Dockerform @Inject constructor(objects: ObjectFactory) : Baseform(obj
                             "$nodeBuildDir/drivers:/opt/corda/drivers"
-
                     ),
                     "ports" to listOf(it.rpcPort, it.config.getInt("sshd.port")),
-                    "image" to "corda/corda-zulu-${it.runtimeVersion().toLowerCase()}"
+                    "image" to (it.dockerImage ?: "corda/corda-zulu-${it.runtimeVersion().toLowerCase()}")
             )
-        }.toMap()
 
+            if (it.dockerConfig.isNotEmpty()) {
+
+                var dockerConfig = ConfigFactory.parseMap(it.dockerConfig)
+
+                val dockerfile = when {
+                    dockerConfig.hasPath("dbDockerConfig.dockerfile") -> dockerConfig.getString("dbDockerConfig.dockerfile")
+                    else -> DEFAULT_DB_DOCKERFILE
+                }
+                val dbInit = when {
+                    dockerConfig.hasPath("dbDockerConfig.dbInit") -> dockerConfig.getString("dbDockerConfig.dbInit")
+                    else -> DEFAULT_DB_INIT_FILE
+                }
+                val dbName = when {
+                    dockerConfig.hasPath("dbDockerConfig.dbName") -> dockerConfig.getString("dbDockerConfig.dbName")
+                    else -> DEFAULT_DB_NAME
+                }
+                val dbSchema = when {
+                    dockerConfig.hasPath("dbDockerConfig.dbSchema") -> dockerConfig.getString("dbDockerConfig.dbSchema")
+                    else -> DEFAULT_DB_SCHEMA
+                }
+                val dbUser = when {
+                    dockerConfig.hasPath("dbDockerConfig.dbUser") -> dockerConfig.getString("dbDockerConfig.dbUser")
+                    else -> DEFAULT_DB_USER
+                }
+                val dbPassword = when {
+                    dockerConfig.hasPath("dbDockerConfig.dbPassword") -> dockerConfig.getString("dbDockerConfig.dbPassword")
+                    else -> DEFAULT_DB_PASSWORD
+                }
+                val dbDataSourceClassName = when {
+                    dockerConfig.hasPath("dataSourceProperties.dataSourceClassName") -> dockerConfig.getString("dataSourceProperties.dataSourceClassName")
+                    else -> DEFAULT_DB_DATA_SOURCE_CLASS_NAME
+                }
+                val dbTransactionIsolationLevel = when {
+                    dockerConfig.hasPath("database.transactionIsolationLevel") -> dockerConfig.getString("database.transactionIsolationLevel")
+                    else -> DEFAULT_DB_TRANSACTION_ISOLATION_LEVEL
+                }
+                val dbRunMigration = when {
+                    dockerConfig.hasPath("database.runMigration") -> dockerConfig.getBoolean("database.runMigration")
+                    else -> DEFAULT_DB_RUN_MIGRATION
+                }
+
+                val dbPort = DEFAULT_DB_STARTING_PORT + index
+                val dbHost = "${it.containerName}-db"
+
+                val dbUrl = when {
+                    dockerConfig.hasPath("dataSourceProperties.dataSource.url") -> {
```

Comment:
There _is_ actually a way to persuade TypeSafe Config to perform these substitutions, but it requires the `DBURL` value to be (err) _"reformatted"_ first... :scream_cat: 

Ideally, we'd like to do something like this:
```kotlin
val fallback = ConfigFactory.empty()
    .withValue("DBHOSTNAME", ConfigValueFactory.fromAnyRef(dbHost))
    .withValue("DBPORT", ConfigValueFactory.fromAnyRef(dbPort))
    .withValue("DBNAME", ConfigValueFactory.fromAnyRef(dbName))
    .withValue("DBSCHEMA", ConfigValueFactory.fromAnyRef(dbSchema))

// val dbUrl = "jdbc:postgresql://\${DBHOSTNAME}:\${DBPORT}/\${DBNAME}?currentSchema=\${DBSCHEMA}"
val config = ConfigFactory.empty()
    .withValue("DBURL", ConfigValueFactory.fromAnyRef(dbUrl))
    .withFallback(fallback)
    .resolve()
```
However, Typesafe doesn't resolve the `${DBNAME}` etc values for us here because `ConfigValueFactory.fromAnyRef()` adds `DBURL` as a `ConfigString$Quoted` object, i.e. a _literal_ string value :facepalm:. The only way around this  (apparently?!)  is to _parse_ the `DBURL` value instead:
```kotlin
val config = ConfigFactory.parseReader(StringReader("DBURL=$dbUrl"))
    .withFallback(fallback)
    .resolve()
```
However, the parser only accepts valid HCON documents, which means that would then need to wrap the literal elements of `dbUrl` in quotes to escape any ":" or "&" characters (as you already discovered).

I think this should be doable...